src/turbomind/models/llama/SequenceManager.h

```diff
@@ -104,13 +109,19 @@ class SequenceManager {
                                       int                          step_length,
                                       AdjustInputCount             adjust);
 
+    void CacheIfEnabled(const Sequence& sequence);
+
     [[nodiscard]] void* GetBlockPtr(int block_id)
     {
         return block_manager_->block(block_id).data;
     }
 
     int max_block_count() const noexcept
     {
+        if (block_trie_->enabled()) {
+            // if enable prefix caching, the max_block_count in batch can be larger
+            return block_manager_->max_block_count() * 2;
```

Comment:
Using `twice` needs to be discussed. @lzhangzz @lvhan028

Reply:
I tested on the sharegpt dataset with 1000 prompts and observed that the peek cumulated block_ptr count in a batch will achieve `1100` when enabling block reuse, and the `max_block_count` is `878`. So I suggest to increase the limitation larger, maybe `2*max_block_count`.

Reply:
> I tested on the sharegpt dataset with 1000 prompts and observed that the peek cumulated block_ptr count in a batch will achieve `1100` when enabling block reuse, and the `max_block_count` is `878`. So I suggest to increase the limitation larger, maybe `2*max_block_count`.

@lzhangzz @lvhan028 @grimoire Do you have any suggestions? Thanks.